### DISCUSSION TITLE
Segmented bar rendering method

### DISCUSSION THREAD

**Discussion by ItsShamed** [ppy/osu#22120] [Q&A] [answered]
I don't know if this is the right place to ask this but, I tried to implement this "segmented bar" thing seen in different places on the Figma design, and I ended up with two ways of rendering it:
- By using a `BufferedContainer` and `Box`es to draw it
- By drawing it up manually using a custom `DrawNode`

Which one would fit this component better?

[![image](https://user-images.githubusercontent.com/39100084/211659042-2e025428-9b07-46fe-8371-c24c727e7cd2.png)](https://www.figma.com/file/DXKwqZhD5yyb1igc3mKo1P/Song-Select-Draft-2?node-id=0%3A1&t=DLW76NzddpUDkD1A-1)

[![image](https://user-images.githubusercontent.com/39100084/211659117-c1124579-daf7-4b81-b90e-34e3875e52d3.png)](https://www.figma.com/file/up6guVobzpc3DZlQxWzVmv/Gameplay-Layout?node-id=0%3A1&t=trSBTqEtzdWEGKaG-1)

**Comment by mk56-spn** [comment]
considering the component the bottom bar is replacing uses a draw node, and that its to my knowledge more efficient  than throwing boxes at the problem im guessing that's likely the appropriate method

**Reply by ItsShamed** [reply]
Yeah to me it seems like the most logical thing to do but I was asking because it *could* have some drawbacks performance wise

**Comment by peppy** [comment, marked as answer]
it's definitely worth avoiding buffered containers if at all possible.

I think it may not even need a custom draw node if sections which are smaller than one display pixel are not drawn (assuming the colours don't flip-flop too often). But if the draw node is simple enough to implement then that's going to be the most performant and flexible.

**Reply by ItsShamed** [reply]
> sections which are smaller than one display pixel are not drawn (assuming the colours don't flip-flop too often) 

Yeah, in theory, it should not face that issue if the input data is good (and small) enough. Though, I may want to limit how much data can be processed.
